web/src/client/storage.js

```diff
@@ -23,147 +23,453 @@
 
 import DBusClient from "./dbus";
 import { WithStatus, WithValidation } from "./mixins";
-import cockpit from "../lib/cockpit";
 
-const STORAGE_SERVICE = "org.opensuse.DInstaller.Storage";
-const STORAGE_PATH = "/org/opensuse/DInstaller/Storage1";
 const PROPOSAL_CALCULATOR_IFACE = "org.opensuse.DInstaller.Storage1.Proposal.Calculator";
+const ISCSI_NODE_IFACE = "org.opensuse.DInstaller.Storage1.ISCSI.Node";
+const ISCSI_NODES_NAMESPACE = "/org/opensuse/DInstaller/Storage1/iscsi_nodes";
+const ISCSI_INITIATOR_IFACE = "org.opensuse.DInstaller.Storage1.ISCSI.Initiator";
 const PROPOSAL_IFACE = "org.opensuse.DInstaller.Storage1.Proposal";
+const STORAGE_OBJECT = "/org/opensuse/DInstaller/Storage1";
 
 /**
- * Storage base client
+ * Removes properties with undefined value
  *
- * @ignore
+ * @example
+ * removeUndefinedCockpitProperties({
+ *  property1: { t: "s", v: "foo" },
+ *  property2: { t: b, v: false },
+ *  property3: { t: "s", v: undefined }
+ * });
+ * //returns { property1: { t: "s", v: "foo" }, property2: { t: "b", v: false } }
+ *
+ * @param {object} cockpitObject
+ * @returns {object}
  */
-class StorageBaseClient {
+const removeUndefinedCockpitProperties = (cockpitObject) => {
```

Comment:
It looks like a generic function that might be useful in other parts. Of course, it can live here by now, but it will be reusable in the future (when we add support for more iSCSI-like APIs). And when we move that function, we could extend it to work on nested objects.